dbscheme-proposal.md

```diff
@@ -0,0 +1,120 @@
+# Database Scheme
+
+## scheme
+* type (enc/sig)
+* name
+* category (lattice etc)
+* exact problems plus type of reduction (list of tuples?)
+* year (what date to use?)
+* inventors ?
+* trust in scheme (how to formalize this?)
+  * attacks
+  * cryptanalysis
+* statefulness (what options are there?)
+
+## parameter set (better name?)
+* -> scheme
+* name ("Lightsaber", "Papa Bear")
+* type/api (better name?) (ktm, kem, ...)
+* security level achieved
+  * classical (bits)
+  * quantum (bits)
+  * nist category (1..5)
+* security notion achieved (IND-CCA, EUF-CMA, ...)
+  * number of operations supported (one-time/few-time/...)
+* sizes (bytes)
+  * public key
+  * secret key (using nist api? (decaps is not given pub key))
+    * also store "real" secret key size?
+  * ciphertext
+  * message (for KEMs, shared secret)
+* failure probability (log2, or 0)
+* diffie-hellman-ness??
+  * symmetry
+  * contributory
+  * static keys
+
+## implementation
+* -> parameter set
+* description ("reference", "optimized using AES")
+* platform (roughly)
+* type of implementation (reference, optimized)
+* security properties
+  * side-channel attacks guarded against
+    * timing
+    * power consumption
+    * ??
+* hardware features used (with respect to timings) (move some of this to timing?)
```

Comment:
I think I'd just add the used CPU features here, i.e. SSE, SSE2, AVX, PCLMUL, etc.